docker_push_latest_if_changed.py

```diff
@@ -1,5 +1,182 @@
 #!/usr/bin/env python3
 import argparse
+import hashlib
+import logging
+from collections import namedtuple
+
+import docker
+
+
+ImageParityFields = namedtuple(
+    'ImageParityFields',
+    ['image_cmds_hash', 'packages_hash'],
+)
+
+GET_PACKAGE_LIST_COMMAND = 'dpkg -l'
+
+
+def _docker_push_latest_if_changed(source, target, is_dry_run):
+    docker_client = _get_docker_client()
+    _validate_source(source, docker_client)
+    target = _get_sanitized_target(source, target)
+
+    logging.info('Pushing source image')
+    _push_image(source, docker_client, is_dry_run)
+    try:
+        logging.warning('Pulling target image')
+        docker_client.pull(target)
+    except docker.errors.NotFound:
+        logging.warning((
+            'Target image {} was not found in the registry. '
+            'Going to attempt to tag and push the target image anyway.'
+        ).format(target))
```

Comment:
probably a good idea to just drop python3.5 and just use python3.6+ and f-strings

Reply:
Sounds good to me
Went gung-ho with the f-strings